frontend/src/metabase/visualizations/visualizations/Table.jsx

```diff
@@ -379,9 +382,14 @@ export default class Table extends Component {
     if (isPivoted) {
       return formatColumn(column) || (columnIndex !== 0 ? t`Unset` : null);
     } else {
-      return (
-        settings.column(column)["_column_title_full"] || formatColumn(column)
-      );
+      const resultsMetadataColumn = this.state.data.resultsMetadataColumns[
+        columnIndex
+      ];
+      const isMetadataDisplayNameOverridden =
+        resultsMetadataColumn.name !== resultsMetadataColumn.display_name;
```

Comment:
I'm not sure if this is a suitable way to check if metadata is overridden, but I couldn't think of any other way. Let me know if this should be done differently.

#### `result_metadata.[index]` from `/api/card/<id>`
![image](https://user-images.githubusercontent.com/1937582/158807349-bb03a2bf-0241-4717-92af-130f6970cbb2.png)